WeightProducer/python/getWeightProducer_cff.py

```diff
@@ -56,7 +56,7 @@ def getWeightProducer(fileName, scan=False):
     for sample in samples:
         if sample.name in fileName and sample.production in fileName:
             weightProducer.Method     = cms.string(sample.Method)
-            weightProducer.XS         = cms.double(sample.XS)
+            weightProducer.XS         = cms.double(sample.XS*sample.kFactor)
```

Comment:
should we consider storing k-factor in a separate branch also?

Reply:
I think you mean as a separate member of the weightProducer and yes, I think that would be prudent. I was going to make the change in a different PR, but I can do it in this one if you'd like. Just let me know.

Reply:
I meant a separate branch in the output tree, but how it's handled in WeightProducer could be modified similarly. It's fine to make another PR for that (in that case, I would leave this one as it currently is, so the meaning of the "cross section" output value is preserved)

Reply:
Okay, I'll leave this alone for now. I'll make a different PR which takes the k-factor out of the XS value and puts it in a separate branch in the tree.